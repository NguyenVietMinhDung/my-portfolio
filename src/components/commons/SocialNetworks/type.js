import type { Props as Item } from './components/SocialNetworkItem/type';

export type Props = {
  items: Array<Item>,
};
