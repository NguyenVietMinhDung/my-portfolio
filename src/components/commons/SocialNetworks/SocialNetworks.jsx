// @flow
import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import styles from './SocialNetworks.scss';
import type { Props } from './type';
import SocialNetworkItem from './components/SocialNetworkItem';

const SocialNetworks = (props: Props) => {
  const { items } = props;
  return (
    <ul styleName="social-networks">
      {items.map(item => (
        <SocialNetworkItem
          key={item.name}
          {...item}
        />
      ))}
    </ul>
  );
};

export default memo<Props>(CSSModules(SocialNetworks, styles));
