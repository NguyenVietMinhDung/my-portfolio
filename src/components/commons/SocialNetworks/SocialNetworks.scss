.social-networks {
	display: flex;
	flex-flow: row wrap;
}
