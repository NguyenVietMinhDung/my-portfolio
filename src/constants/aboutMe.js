export const TITLE = 'About Me';
export const GREETING = 'Hi There! I\'m Dung Nguyen';
export const SUMMARY = 'I am an ambitious Web\n'
  + 'Application Developer who\n'
  + 'always tries to improve himself\n'
  + 'everyday, a quick learner who\n'
  + 'can absorb new ideas and can\n'
  + 'communicate clearly and\n'
  + 'effectively. Looking for an\n'
  + 'ambitious company which will\n'
  + 'challenge my developer and\n'
  + 'allows me to continue\n'
  + 'to develop my knowledge and\n'
  + 'potential.';
export const basicInfo = [
  {
    iconClassName: 'lni-calendar',
    name: 'Date of birth',
    value: '16 January 1995',
  },
  {
    iconClassName: 'lni-envelope',
    name: 'Email',
    value: 'ngvietminhdung@gmail.com',
  },
  {
    iconClassName: 'lni-phone-handset',
    name: 'Phone',
    value: '(076) 401-2912',
  },
  {
    iconClassName: 'lni-map-marker',
    name: 'Location',
    value: 'Tan Binh District, HCM City',
  },
];
export const skills = [
  {
    itemId: 'skill-1',
    name: 'JavaScript, HTML, CSS',
    value: 80,
  },
  {
    itemId: 'skill-2',
    name: 'ReactJS',
    value: 60,
  },
  {
    itemId: 'skill-3',
    name: 'SCSS',
    value: 60,
  },
  {
    itemId: 'skill-5',
    name: 'Webpack',
    value: 50,
  },
  {
    itemId: 'skill-6',
    name: 'Node.js',
    value: 60,
  },
  {
    itemId: 'skill-7',
    name: 'Git',
    value: 60,
  },
];
