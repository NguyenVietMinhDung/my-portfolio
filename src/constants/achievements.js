export const TITLE = 'Achievements';
export const timeline = {
  iconClassName: 'lni-certificate',
  items: [
    {
      key: 1,
      organization: 'Amazon Web Services (AWS)',
      startDate: 'August 2020',
      endDate: 'August 2023',
      description: 'AWS Certified Cloud Practitioner',
    },
  ],
};
