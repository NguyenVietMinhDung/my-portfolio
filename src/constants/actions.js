export const ACTIVATE_NAVIGATION_ITEM = 'ACTIVATE_NAVIGATION_ITEM';
export const OPEN_NAVIGATION_MENU = 'OPEN_NAVIGATION_MENU';
