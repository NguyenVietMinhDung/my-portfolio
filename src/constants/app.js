import facebookIconUrl from '../assets/images/facebook.png';
import instagramIconUrl from '../assets/images/instagram.png';
import linkedInIconUrl from '../assets/images/linkedin.png';

export const LOADER_SIZE = 50;
export const LOADER_SIZE_UNIT = 'px';
export const LOADER_COLOR = '#754ef9';
export const socialNetworks = [
  {
    name: 'Facebook',
    url: 'https://www.facebook.com/minh.dung.391',
    iconUrl: facebookIconUrl,
  },
  {
    name: 'Instagram',
    url: 'https://www.instagram.com/dungvmnguyen',
    iconUrl: instagramIconUrl,
  },
  {
    name: 'LinkedIn',
    url: 'https://www.linkedin.com/in/dungvmn1995',
    iconUrl: linkedInIconUrl,
  },
];
export logoUrl from '../assets/images/logo.png';
