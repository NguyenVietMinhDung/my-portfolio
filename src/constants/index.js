import * as actions from './actions';
import * as app from './app';
import * as navbar from './navbar';
import * as header from './header';
import * as footer from './footer';
import * as aboutMe from './aboutMe';
import * as experiences from './experiences';
import * as educations from './educations';
import * as projects from './projects';

export default {
  actions,
  app,
  navbar,
  header,
  footer,
  aboutMe,
  experiences,
  educations,
  projects,
};
