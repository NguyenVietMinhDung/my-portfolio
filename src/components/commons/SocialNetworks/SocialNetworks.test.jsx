import React from 'react';
import { shallow } from 'enzyme';
import { socialNetworks as items } from '../../../constants/app';
import SocialNetworks from './SocialNetworks';

const defaultProps = {
  items,
};

describe('<SocialNetworks />', () => {
  test('renders correctly', () => {
    const component = shallow(<SocialNetworks {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
