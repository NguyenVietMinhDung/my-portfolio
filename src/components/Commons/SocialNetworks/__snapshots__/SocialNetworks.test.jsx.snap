// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialNetworks /> renders correctly 1`] = `
<ul
  className="social-networks"
>
  <Memo(SocialNetworkItem)
    iconUrl="test-file-stub"
    key="Facebook"
    name="Facebook"
    url="https://www.facebook.com/minh.dung.391"
  />
  <Memo(SocialNetworkItem)
    iconUrl="test-file-stub"
    key="Instagram"
    name="Instagram"
    url="https://www.instagram.com/dungvmnguyen"
  />
  <Memo(SocialNetworkItem)
    iconUrl="test-file-stub"
    key="LinkedIn"
    name="LinkedIn"
    url="https://www.linkedin.com/in/dungvmn1995"
  />
</ul>
`;
